babel/numbers.py

```diff
@@ -416,15 +429,54 @@ def format_decimal(
                                  the format pattern. Defaults to `True`.
     :param group_separator: Boolean to switch group separator on/off in a locale's
                             number format.
+    :param format_type: Format to use (`None`, "short" or "long"). The standard decimal
+                        format is `None`. Defaults to `None`.
+    :param compact_fraction_digits: Number of fraction digits to use in "short" or "long"
+                                    format. Defaults to `0`. If this is set to
+                                    a value greater than `0`, the `decimal_quantization`
+                                    will be treated as `False`.
     """
     locale = Locale.parse(locale)
-    if not format:
-        format = locale.decimal_formats.get(format)
+    if format_type in ("short", "long"):
+        number, format = _get_compact_format(number, format_type, locale, compact_fraction_digits)
+        # if compact_fraction_digits is set, we don't want to truncate the fraction digits
+        decimal_quantization = False if compact_fraction_digits > 0 else decimal_quantization
```

Comment:
I'd say just use a good old `if` – and no need to look at whether the value is strictly positive either; we just care about it being or not being set.

```suggestion
        # if compact_fraction_digits is set, we don't want to truncate the fraction digits
        if compact_fraction_digits:
            decimal_quantization = False
```